sigal/gallery.py

```diff
@@ -479,7 +489,11 @@ def zip(self):
         zip_gallery = self.settings['zip_gallery']
 
         if zip_gallery and len(self) > 0:
+            zip_gallery = zip_gallery.format(album=self)
             archive_path = join(self.dst_path, zip_gallery)
+            if isfile(archive_path):
+                self.logger.debug("Archive %s already created, passing", archive_path)
+                return zip_gallery
```

Comment:
This will not give the expected result if images are added to an album. Sometimes it seems reasonable to skip some operation (for the big image case, we already suppose that images are not modified, so the thumbnails are not recomputed) but here it breaks a common use case.